src/components/AbuseReportForm/AbuseReportForm.tsx

```diff
@@ -63,100 +94,154 @@ export const Form: React.FC<{
+    };
 
-    {errors && errors.response && (
-      <span
-        className={css`
-          color: red;
-        `}
-      >
-        {errors.response}
-      </span>
-    )}
-
-    <div className={inputWrapper}>
-      <label htmlFor="category">Category</label>
-      <select
-        name="category"
-        id="category"
-        onChange={categoryOnChange}
-        value={selectedCategory}
-      >
-        <option selected value="0">
-          Please select
-        </option>
-        <option value="1">Personal abuse</option>
-        <option value="2">Off topic</option>
-        <option value="3">Legal issue</option>
-        <option value="4">Trolling</option>
-        <option value="5">Hate speech</option>
-        <option value="6">Offensive/Threatening language</option>
-        <option value="7">Copyright</option>
-        <option value="8">Spam</option>
-        <option value="9">Other</option>
-      </select>
-      {errors && errors.category && (
-        <span
-          className={css`
-            color: red;
-          `}
-        >
-          {errors.category}
-        </span>
-      )}
-    </div>
+    document.addEventListener("keydown", keyListener);
 
-    <div className={inputWrapper}>
-      <label htmlFor="reason">Reason (optional)</label>
-      <textarea
-        name="reason"
-        onChange={reasonOnChange}
-        value={reasonText}
-      ></textarea>
-    </div>
+    return () => document.removeEventListener("keydown", keyListener);
+  });
 
-    <div className={inputWrapper}>
-      <label htmlFor="email">Email (optional)</label>
-      <input
-        type="email"
-        name="email"
-        onChange={emailOnChange}
-        value={emailText}
-      ></input>
-    </div>
+  const modalRef = createRef<HTMLDivElement>();
+  useEffect(() => {
+    if (!modalRef.current) return;
+    const closeModal = modalRef.current.querySelector(
+      'select[name="category"]'
+    );
+    closeModal && closeModal.focus();
+  }, [modalRef]);
 
-    <div
-      className={css`
-        float: right;
-      `}
-    >
-      <Button onClick={submitForm} type="submit" size="small">
-        Report
-      </Button>
+  const handleTabKey = (e: KeyboardEvent) => {
+    if (!modalRef.current) return;
+    const focusableModalElements = modalRef.current.querySelectorAll(
+      'select[name="category"], textarea[name="reason"], input[type="email"], button[type="submit"], button[custom-guardian="close-modal"]'
+    );
+    const firstElement = focusableModalElements[0];
+    const lastElement =
+      focusableModalElements[focusableModalElements.length - 1];
+
+    if (!e.shiftKey && document.activeElement === lastElement) {
+      firstElement.focus();
+      return e.preventDefault();
+    }
+
+    if (e.shiftKey && document.activeElement === firstElement) {
+      lastElement.focus();
+      e.preventDefault();
+    }
+  };
+
+  const keyListenersMap = new Map([
+    [27, toggleSetShowForm],
```

Comment:
The `esc` key binding here is a nice touch! Maybe you could set focus on the report button so the user is back to where they might expect when the form is closed (either by esc or close button)